### ISSUE TITLE
server could generate its own ticket strings

### ISSUE THREAD

**Issue by d-w-moore** [irods/irods#5990]
## Feature

Currently client applications are allowed to generate their own strings to assert the uniqueness of a ticket whose creation they are requesting. The Python iRODS Client [does this](https://github.com/irods/python-irodsclient/blob/95096356c9e493f37433dfba37d2a11be4f9afa4/irods/ticket.py#L15).

This approach could be problematic because:
   - it opens the door to different clients' requests colliding though these are less likely when randomly generated. 
   - it could be a security issue that clients are allowed to specify this string at all, as a rogue user could spoof another ticket's identity as soon as it is deleted from the catalog

**Comment by trel** [comment]
Right, I don't see much value in having the client be able to request a particular string.

Is there a use case where this is ... essential?

**Comment by d-w-moore** [comment]
Some might like tickets to have mnemonic value or be generated from a hash value, but if they want that, the server generated ticket string can be stored in a map structure under the purpose-appropriate key, I think....

We could require any client-generated strings to be of the same length as the ticket string column allows.

Or as a hybrid approach,... a bit far-out perhaps: What we currently think of as the client generated string might be interpreted as a **_template_** instead, padded (on the right side) with asterisks to fill it out to the DB column width.  Each asterisk would be replaced with a random server-generated character to hugely reduce collision probability. We might specify in the API that at most half the template characters can be client-specified (ie anything not an asterisk).

**Comment by trel** [comment]
The template/prefix approach is ... messy?   Doesn't provide the client with deterministic behavior - in either direction.

I think the client-generated approach is fine... but when a collision happens, the server has no recourse other than a new server-generated string... or an error of 'already exists'... neither is very great.

If we deprecate the client-specified approach... we can guarantee a unique ticket string will be returned every time one is requested (or something else went wrong).

I think 'token-based' access is a lot more common/familiar today than a few years ago.  Having these strings hold semantic or mnemonic meaning feels dated ... determinism feels more correct here.
